File: test_data/code/java/1018-binary-prefix-divisible-by-5.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> answer = new ArrayList<>();
        int sum = 0;
        for (int num : nums) {
            sum = (sum * 2 + num) % 5;
            answer.add(sum == 0);
        }
        return answer;
    }
}

File: test_data/code/python/1018-binary-prefix-divisible-by-5.py
from typing import List

class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        res = []
        temp = 0
        for num in nums:
            temp = (temp * 2 + num) % 5
            res.append(temp == 0)
        return res